### ISSUE TITLE
[Audio] Toggle setting the currently playing song as a voice channel status

### ISSUE THREAD

**Issue by EternalllZM** [Cog-Creators/Red-DiscordBot#6442]
### What component of Red (cog, command, API) would you like to see improvements on?

Audio

### Describe the enhancement you're suggesting.

Current functionality of audioset status according to docs:
[p]audioset status - Show the now playing song in the bot’s status, or show how many servers the bot is playing music on, if more than one.

This command and functionality could be extended (sub command?) to also display the currently playing song as a voice channel status.

![image](https://github.com/user-attachments/assets/9a4a4efe-1bb5-4947-a740-8979f6250d6e)

### Anything else?

While the last comment on this issue mentions channel status may be replaced by hang status - recently the hang status was completely rolled back to 0% and cements channel status as what is likely staying instead https://github.com/discord/discord-api-docs/pull/6398. 

Unfortunately I couldn't find anything showing this endpoint been implemented yet, so even if this idea is approved, it may have to stay blocked for a while.

**Comment by Jackenmen** [comment]
It is currently unclear what Discord will do with this - this feature went through 3 different API implementations, I believe that the feature is presently in A/B testing in clients (i.e. only present for *some* users) and not too long ago there was a similar feature that allowed individual members to set voice statuses but seems to no longer be available (at least for me).
We'll have to see what this settles on first. Presumably, it won't be implemented in our upstream library before that time either so it will definitely be blocked for a long while.

**Comment by bryceoj** [comment]
Any update on this whether this is something that can implemented?  

Interested in having audio update the voice channel status to the current song playing.  Should be easily implemented alongside the existing audio status for the bot's status, but instead change the voice channel status.

Using the channel.edit:
https://discordpy.readthedocs.io/en/latest/api.html?highlight=voice%20channel#discord.VoiceChannel.edit

**Comment by Flame442** [comment]
This seems like a reasonable feature if someone is interested in developing it. Not sure if it would be better to make this an alternative option within `[p]audioset status` (similar to `[p]set usemenus`) or as an entirely separate toggle so both could be enabled. The main downside to using the bot status is that it is a global setting, so this would allow similar functionality for all servers the bot is playing in.

**Comment by bryceoj** [comment]
@Flame442 yeah even just an additional toggle separate from `[p]audioset status`  could easily just be `[p]audioset voicestatus`

I would easily attempt to make a PR for this change, just let me know.

**Comment by bryceoj** [comment]
@Flame442 started working on this. 
https://github.com/bryceoj/Red-DiscordBot/tree/audio-status-voice-channel

**Comment by EternalllZM** [comment]
Per Discord's API announcement today [we now have documentation](https://docs.discord.com/developers/change-log#voice-channel-status-and-start-time-documentation) about voice channel status and start time. 

- [Set Voice Channel Status](https://docs.discord.com/developers/resources/channel#set-voice-channel-status) endpoint: Set a short description (up to 500 characters) for a voice channel. Requires the  SET_VOICE_CHANNEL_STATUS permission (1 << 48), and MANAGE_CHANNELS if your bot isn't connected to the channel
- Gateway commands & events: Use the Request Channel Info command (opcode 43) to read status and voice_start_time, and listen for Voice Channel Status Update and Voice Channel Start Time Update events.
- Audit log events: VOICE_CHANNEL_STATUS_UPDATE (192) and VOICE_CHANNEL_STATUS_DELETE (193) for tracking status changes.
